pkg/kubectl/cmd/util/helpers.go

```diff
@@ -332,8 +333,28 @@ func AddValidateFlags(cmd *cobra.Command) {
 	cmd.MarkFlagFilename("schema-cache-dir")
 }
 
-func AddRecursiveFlag(cmd *cobra.Command, value *bool) {
-	cmd.Flags().BoolVarP(value, "recursive", "R", *value, "If true, process directory recursively.")
+func addRecursiveFlag(cmd *cobra.Command, options *resource.FilenameParamOptions) {
+	cmd.Flags().BoolVarP(&options.Recursive, "recursive", "R", options.Recursive, "If true, process directory recursively.")
+}
+
+func addHttpRetryFlag(cmd *cobra.Command, options *resource.FilenameParamOptions) {
+	cmd.Flags().IntVar(&options.FilenameHttpRetries, FilenameHttpRetries, 1, "Retry failed http requests for urls passed to '-f' flags this many times.")
+}
+
+func addFlagJsonFilenameFlag(cmd *cobra.Command, options *resource.FilenameParamOptions, usage string) {
+	kubectl.AddJsonFilenameFlag(cmd, &options.Filenames, usage)
+}
+
+func AddFilenameParamFlags(cmd *cobra.Command, options *resource.FilenameParamOptions, usage string) {
+	addFlagJsonFilenameFlag(cmd, options, usage)
```

Comment:
Why not directly call `kubectl.AddJsonFilenameFlag(cmd, &options.Filenames, usage)` here?